base/array.jl

```diff
@@ -27,11 +27,12 @@ import Core: arraysize, arrayset, arrayref
 
 """
     Array{T}(dims)
+    Array{T,N}(dims)
```

Comment:
> we use brackets in lots of other places in the documentation for optional arguments

And there's been a gradual move towards getting rid of them and using actual Julia syntax in more places. The suggestion with `...` seems clear enough - also not exactly valid syntax but obvious what it means.